tree/dataframe/CMakeLists.txt

```diff
@@ -37,6 +37,7 @@ ROOT_STANDARD_LIBRARY_PACKAGE(ROOTDataFrame
                               SOURCES ${sources}
                               DICTIONARY_OPTIONS "-writeEmptyRootPCM"
                               LIBRARIES ${TBB_LIBRARIES} ${ARROW_SHARED_LIB} ${DATAFRAME_SQLITE_LIBRARIES}
+			      ${SQLITE_LIBRARIES} Davix::Davix
```

Comment:
I think we need to adjust that.  In a first step, we should depend on Davix only if the sqlite datasource is actually build, like

- Remove `${SQLITE_LIBRARIES} Davix::Davix` here
- Add `Davix::Davix` [before|https://github.com/root-project/root/blob/eecb9f2fd8f3e0087958e238cefb10a030680f16/tree/dataframe/CMakeLists.txt#L12] like `set (DATAFRAME_SQLITE_LIBRARIES ${SQLITE_LIBRARIES} Davix::Davix)`

Then we need to think further to deal with the case where sqlite is there but Davix is not.  In this case, we do not want to depend on the `Davix::Davix` library and we want to know this in RSqliteDS.cxx so that we can avoid compiling the custom vfs layer.  About this second step, I'd still need to think a little.